debezium-core/src/main/java/io/debezium/transforms/outbox/EventRouter.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms.outbox;
+
+import io.debezium.config.Configuration;
+import io.debezium.data.Envelope;
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.transforms.ExtractField;
+import org.apache.kafka.connect.transforms.RegexRouter;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+/**
+ * Debezium Outbox Transform Event Router
+ *
+ * @author Renato mefi (gh@mefi.in)
+ */
+public class EventRouter<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventRouter.class);
+
+    private final ExtractField<R> afterExtractor = new ExtractField.Value<>();
+    private final RegexRouter<R> regexRouter = new RegexRouter<>();
+    private EventRouterConfigDefinition.InvalidOperationBehavior invalidOperationBehavior;
+
+    private String fieldEventId;
+    private String fieldEventKey;
+    private String fieldEventType;
+    private String fieldPayload;
+    private String fieldPayloadId;
+    private String fieldPayloadType;
```

Comment:
What is meaning of "payloadType"?

Reply:
So this is the "generic" name for the aggregatetype, the default value of this field is `aggregatetype` as from the outbox blog post

Reply:
Gotcha. Thanks for clarifying.